src/platform/silabs/wifi/DnssdImpl.cpp

```diff
@@ -0,0 +1,172 @@
+    }
+}
+
+const char * GetProtocolString(DnssdServiceProtocol protocol)
+{
+    return protocol == DnssdServiceProtocol::kDnssdProtocolTcp ? "_tcp" : "_udp";
+}
+
+sl_mdns_service_t mdnsService = {};
+std::string serviceMessage;
+
+CHIP_ERROR ChipDnssdPublishService(const DnssdService * service, DnssdPublishCallback callback, void * context)
+{
+    sl_mdns_service_t mdnsService = {};
+    std::string serviceMessage;
+
+    char service_type[256]; // Allocate enough memory for the concatenated string
+    snprintf(service_type, sizeof(service_type), "%s.%s.local.", service->mType, GetProtocolString(service->mProtocol));
+
+    char instance_name[512]; // Allocate enough memory for the concatenated string
+    snprintf(instance_name, sizeof(instance_name), "%s.%s.%s.local", service->mName, service->mType,
+             GetProtocolString(service->mProtocol));
+
+    mdnsService.instance_name = strdup(instance_name); // Duplicate the string to ensure memory safety
+    mdnsService.service_type  = strdup(service_type);  // Duplicate the string to ensure memory safety
+    mdnsService.port          = service->mPort;
+    mdnsService.ttl           = 120; // Default TTL; adjust as needed
+    if (service->mTextEntries && service->mTextEntrySize > 0)
+    {
+        for (size_t i = 0; i < service->mTextEntrySize; i++)
+        {
+            if (service->mTextEntries[i].mKey == nullptr || service->mTextEntries[i].mData == nullptr)
+            {
+                continue;
+            }
+            serviceMessage += std::string(service->mTextEntries[i].mKey) + "=" +
+                std::string(reinterpret_cast<const char *>(service->mTextEntries[i].mData)) + " ";
+        }
+        mdnsService.service_message = strdup(serviceMessage.c_str()); // Duplicate the string to ensure memory safety
+    }
+    sl_status_t status = sl_mdns_register_service(&gMdnsInstance, SL_NET_WIFI_CLIENT_INTERFACE, &mdnsService);
+    if (status != SL_STATUS_OK)
+    {
+        ChipLogError(DeviceLayer, "Failed to publish service: 0x%lx", status);
+        return CHIP_ERROR_INTERNAL;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ChipDnssdRemoveServices()
+{
+    // Currently, `sl_mdns_unregister_service` is not supported in `sl_mdns.c`.
+    // As a workaround, we can deinitialize and reinitialize the mDNS instance to remove all services.
+    sl_mdns_deinit(&gMdnsInstance);
+
+    sl_mdns_configuration_t config = { .protocol = SL_MDNS_PROTO_UDP, .type = SL_IPV6_VERSION, .host_name = "chip-device.local" };
+
+    sl_status_t status = sl_mdns_init(&gMdnsInstance, &config, nullptr);
+    if (status != SL_STATUS_OK)
+    {
+        ChipLogError(DeviceLayer, "Failed to reinitialize mDNS: %ld", status);
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    status = sl_mdns_add_interface(&gMdnsInstance, SL_NET_WIFI_CLIENT_INTERFACE);
+    if (status != SL_STATUS_OK)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
```

Comment:
We shouldn't copy paste the code from the init function. We should just call the init function directly with the correct paremeters for the behavior we want.

Reply:
removed.